examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -415,6 +415,75 @@ function interleaveAttributes( attributes ) {
 
 }
 
+// returns a new, non-interleaved version of the provided attribute
+export function deinterleaveAttribute( attribute ) {
+
+	const cons = attribute.data.array.constructor;
+	const count = attribute.count;
+	const itemSize = attribute.itemSize;
+	const normalized = attribute.normalized;
+
+	const array = new cons( count * itemSize );
+	let newAttribute;
+	if ( attribute.isInstancedInterleavedBufferAttribute ) {
+
+		newAttribute = new InstancedBufferAttribute( array, itemSize, normalized, attribute.meshPerAttribute );
+
+	} else {
+
+		newAttribute = new BufferAttribute( array, itemSize, normalized );
+
+	}
+
+	for ( let i = 0; i < count; i ++ ) {
+
+		newAttribute.setX( i, attribute.getX( i ) );
+
+		if ( itemSize >= 2 ) {
+
+			newAttribute.setY( i, attribute.getY( i ) );
+
+		}
+
+		if ( itemSize >= 3 ) {
+
+			newAttribute.setZ( i, attribute.getZ( i ) );
+
+		}
+
+		if ( itemSize >= 4 ) {
+
+			newAttribute.setW( i, attribute.getW( i ) );
+
+		}
+
+	}
+
+	return newAttribute;
+
+}
+
+// deinterleaves all attributes on the geometry
+export function deinterleaveGeometry( geometry ) {
+
+	for ( const key in geometry.attributes ) {
+
+		if ( geometry.attributes[ key ].isInterleavedBufferAttribute ) {
+
+			geometry.attributes[ key ] = deinterleaveAttribute( geometry.attributes[ key ] );
+
+		}
+
+	}
+
+	if ( geometry.index && geometry.index.isInterleavedBufferAttribute ) {
+
+		geometry.index = deinterleaveAttribute( geometry.index );
+
+	}
```

Comment:
I don't think interleaved indices are a thing ([source](https://stackoverflow.com/questions/11148567/rendering-meshes-with-multiple-indices/11148568#11148568)), so we could skip a copy here.

Reply:
gotcha -- done!